site/app/libraries/FileUtils.php

```diff
@@ -519,14 +568,19 @@ public static function validateUploadedFiles($files) {
                 $err_msg = "Invalid filename";
             }
 
-            $success = $err_msg === "No error.";
+            //if zip file check files inside
+            if($is_zip && !FileUtils::checkFileInZipName($tmp_name) ){
+                $err_msg = "Invalid filename within zip file"; 
+            }
 
+            $success = $err_msg === "No error.";
```

Comment:
Let's inline this check into the `$ret[]` definition. No need in creating this temporary variable.